### ISSUE TITLE
Save correct token confidence

### ISSUE THREAD

**Issue by mshannon-sil** [sillsdev/silnlp#949]
Currently, there's two problems with the values being saved for the token confidences. One is that they are raw logits rather than softmaxed logits for BeamSearchEncoderDecoderOutput. The second is a larger issue, which is that there seems to be a logic problem where it's choosing token scores from the wrong beam. I can tell the token score is wrong because in a test case, the end of sentence token was listed as an extremely small confidence of 0.0000058, even though it should have been much more confident if that was the token selected, and there's multiple other places with that kind of issue. The code for finding token scores was taken directly from the machine.py implementation in `hugging_face_nmt_engine.py`, so it's worth investigating whether the same issue occurs in machine.py or if the SILNLP and machine.py implementations diverged in some way.
